src/ldap.rs

```diff
@@ -719,4 +721,18 @@ impl Ldap {
             .await
             .map(|_| ())?)
     }
+
+    /// Returns the TLS peer certificate in the DER format.
+    /// The method returns Ok(None) if no certificate was found or
+    /// if the connection does not use TLS.
+    #[cfg(any(feature = "tls-native", feature = "tls-rustls"))]
+    pub async fn get_peer_certificate_der(&mut self) -> Result<Option<Vec<u8>>> {
+        let (tx, rx) = oneshot::channel();
+        self.misc_tx.send(MiscSender::Cert(tx))?;
+        Ok(if let Some(timeout) = self.timeout.take() {
+            time::timeout(timeout, rx).await?
+        } else {
```

Comment:
No timeout should be needed for inter-task communication. Moreover, the channel is unbounded and isn't going to delay sending any message.